modules/glean/runtime/glean-plugin.client.ts

```diff
@@ -33,24 +33,41 @@ export default defineNuxtPlugin((nuxtApp) => {
       }
     }
 
+    function recordEngagement(element: Element) {
+      if (!element)
+        return
+
+      if (!element.hasAttribute('data-glean'))
+        return
+
+      const data = element.getAttribute('data-glean') || ''
+      const value = element.getAttribute('data-glean-value') || ''
+      engagement.record({ ui_identifier: data, engagement_value: value, ...engagementDetails[data] })
+    }
+
     function handleButtonClick(ev: MouseEvent) {
       const eventTarget = ev?.target as Element
       const closestButton = eventTarget.closest('button')
 
+      if (!closestButton)
```

Comment:
This guard is a functional change. Currently, `handleButtonClick` captures engagement events on any element. Adding this guard is needed to prevent duplicate events from both here and `handleLinkClick` for clicks on links.

cc @wtfluckey

Reply:
love it